File: example/login_test.js
var chai = require("chai");
var chaiAsPromised = require("chai-as-promised");
chai.use(chaiAsPromised);
chai.should();
var wd = require('wd');
var path = require('path')
var apk_path = path.join(process.cwd(), "app-debug.apk")
var keys = require(path.join(process.cwd(), 'keys.json'))

// uploading apk to Sauce Labs
console.log("Uploading apk to Sauce Labs, this may take a while!")
var uploader = require("../dist/index.js");
var result = uploader.uploadSync({user:keys["user"], access_key:keys["accessKey"], app_path:apk_path});
if(result[0] != null) {
  console.log(result[0]);
  return;
}
console.log(result[1]);
var app = ["sauce-storage", result[1]["filename"].trim()].join(":")
var caps = {
  "deviceName":"Android",
  "host":"ondemand.saucelabs.com",
  "port":80,
  "app" : app,
  "username" : keys["user"],
  "accessKey": keys["accessKey"],
  "app-package":"com.urucas.kriket",
  "app-wait-activity": "com.urucas.kriket.activities.LoginActivity",
  "browserName" : "",
  "platformName":"Android",
  "deviceName": "Android Emulator",
  "platformVersion": "5.0", 
  "appium-version" : "1.4.7"
}
chaiAsPromised.transferPromiseness = wd.transferPromiseness;

// run appium test
describe('Test kriket login', function() {
  this.timeout(0);

  describe("Checking all elements are displayed", function() {
    var browser;
    
    before(function() {
      browser = wd.promiseChainRemote(caps);
      return browser
        .init(caps);
    });

    after(function() {
      return browser
        .quit();
    });
    
    it("should find email & password edtiText", function(done) {
      browser
      .elementById("userEmail")
      .isDisplayed()
      .then(function(isDisplayed){
        isDisplayed.should.equal(true)
      })
      .elementById("userEmail")
      .sendKeys("tests@test.com")
      .text()
      .then(function(text){
        text.should.equal("tests@test.com")
      })
      .elementById("userPass")
      .isDisplayed()
      .then(function(isDisplayed){
        isDisplayed.should.equal(true)
      })
      .elementById("userPass")
      .type("pass")
      .hideKeyboard()
      .elementById("loginBtt")
      .isDisplayed()
      .then(function(isDisplayed){
        isDisplayed.should.equal(true)
      })
      .nodeify(done);
    });
     
  });

});

File: example/python/login_test.py
import os
from time import sleep

import unittest
import argparse
import json
import re
import subprocess
from appium import webdriver
import sys

PATH = lambda p: os.path.abspath(
    os.path.join(os.path.dirname(__file__), p)
)

class SauceUploadTests(unittest.TestCase):

    def setUp(self):
        app_path  = PATH("../mocha/app-debug.apk")
        keys_path = PATH("../mocha/keys.json")
        with open(keys_path) as data:
            keys = json.load(data)

        # upload ap using sauce-uploader cli
        result = subprocess.check_output(["sauce-uploader", keys["user"], keys["accessKey"], app_path])
        response = json.loads(result)
        try:
            filename = response["filename"]
        except Exception, e:
            print e
            print result
            self.driver.quit()

        desired_caps =  {
            "deviceName":"Android",
            "host":"ondemand.saucelabs.com",
            "port":80,
            "app" : "sauce-storage:%s" % filename,
            "username" : keys["user"],
            "accessKey": keys["accessKey"],
            "app-package":"com.urucas.kriket",
            "appWaitActivity": "com.urucas.kriket.activities.LoginActivity",
            "browserName" : "",
            "platformName":"Android",
            "deviceName": "Android Emulator",
            "platformVersion": "5.0",
            "appium-version" : "1.4.7"
        }
        self.driver = webdriver.Remote("http://ondemand.saucelabs.com:80/wd/hub", desired_caps)

    def tearDown(self):
        # end the session
        self.driver.quit()

    def test_login(self):
        sleep(.5)
        el = self.driver.find_element_by_id("userEmail")
        self.assertIsNotNone(el)
        self.assertTrue(el.is_displayed())
        el.send_keys('test@test.com')
        text = el.get_attribute("text")
        self.assertEqual('test@test.com', text)

        el = self.driver.find_element_by_id("userPass")
        self.assertIsNotNone(el)
        self.assertTrue(el.is_displayed())

        el = self.driver.find_element_by_id("loginBtt")
        self.assertIsNotNone(el)
        self.assertTrue(el.is_displayed())


if __name__ == '__main__':
    # run test
    suite = unittest.TestLoader().loadTestsFromTestCase(SauceUploadTests)
    unittest.TextTestRunner(verbosity=2).run(suite)

